lib/pages/bootstrap/tom_bootstrap_dialog.dart

```diff
@@ -0,0 +1,223 @@
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/domain/model/recovery_words/recovery_words.dart';
+import 'package:fluffychat/domain/usecases/save_recovery_words_interactor.dart';
+import 'package:fluffychat/widgets/adaptive_flat_button.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:matrix/encryption.dart';
+import 'package:matrix/encryption/utils/bootstrap.dart';
+import 'package:matrix/matrix.dart';
+
+class TomBootstrapDialog extends StatefulWidget {
+  final bool wipe;
+  final Client client;
+  final RecoveryWords? recoveryWords;
+  const TomBootstrapDialog({
+    Key? key,
+    this.recoveryWords,
+    this.wipe = false,
+    required this.client,
+  }) : super(key: key);
+
+  Future<bool?> show(BuildContext context) => showDialog(
+    context: context,
+    builder: (context) => this,
+    barrierDismissible: true,
+    useRootNavigator: false,
+  );
+
+  @override
+  TomBootstrapDialogState createState() => TomBootstrapDialogState();
+}
+
+class TomBootstrapDialogState extends State<TomBootstrapDialog> {
+
+  final _saveRecoveryWordsInteractor = getIt.get<SaveRecoveryWordsInteractor>();
+  late Bootstrap bootstrap;
+  String? titleText;
+
+  UploadRecoveryKeyState _uploadRecoveryKeyState = UploadRecoveryKeyState.initial;
+
+  bool? _wipe;
+
+  @override
+  void initState() {
+    _createBootstrap(widget.wipe);
+    super.initState();
+  }
+
+  void _createBootstrap(bool wipe) async {
+    _wipe = wipe;
+    titleText = null;
+    _uploadRecoveryKeyState = UploadRecoveryKeyState.initial;
+    bootstrap =
+        widget.client.encryption!.bootstrap(onUpdate: (_) => setState(() {}));
```

Comment:
should wrap setState in `if (mounted)` because it is called in initState, so the setState maybe not available yet. And also why this method have to async ?

Reply:
ok, no problem. But it is the callback function